ddtrace/tracer/tracer_test.go

```diff
@@ -388,14 +395,144 @@ func TestSamplingDecision(t *testing.T) {
 		child := tracer.StartSpan("name_2", ChildOf(parent.context)).(*span)
 		child.Finish()
 		parent.Finish()
-		tracer.pushTrace(&finishedTrace{spans: []*span{parent, child}})
 		tracer.Stop()
 		assert.Equal(t, float64(ext.PriorityAutoReject), parent.Metrics[keySamplingPriority])
 		assert.Equal(t, decisionDrop, parent.context.trace.samplingDecision)
 		assert.Equal(t, 8.0, parent.Metrics[keySpanSamplingMechanism])
 		assert.Equal(t, 1.0, parent.Metrics[keySingleSpanSamplingRuleRate])
 		assert.Equal(t, 15.0, parent.Metrics[keySingleSpanSamplingMPS])
 	})
+
+	t.Run("client_dropped_with_single_span_rules", func(t *testing.T) {
+		os.Setenv("DD_SPAN_SAMPLING_RULES", `[{"service": "match","name":"nothing", "sample_rate": 1.0, "max_per_second": 15.0}]`)
+		defer os.Unsetenv("DD_SPAN_SAMPLING_RULES")
+		// Stats are enabled, rules are available, but match nothing, sample rate equals 0.
```

Comment:
It looks like stats aren't enabled for this test?

Reply:
That is correct, will fix the comment